### PR TITLE
Add support for hostname lookups instead of IPs in MINIO_PUBLIC_IPS

### LINKED ISSUE #6862: MINIO_PUBLIC_IPS doesn't accept SRV records
MINIO_PUBLIC_IPS requires the public IPs of the Minio cluster and doesn't accept SRV records or domains. knowing that, In CaaS based environment, the IPs are automatically assigned by docker, kubernetes, etc ...

## Current Behavior
MINIO_PUBLIC_IPS does not accept SRV records and through the following error "ERROR Unable to initialize Minio server with invalid MINIO_PUBLIC_IPS[0]: minio-cluster1: Invalid arguments specified"

## Possible Solution
MINIO_PUBLIC_IPS should accept SRV records or a new environment variable should be introduced.

## Your Environment
docker version
Client: Docker Engine - Community
Version: 18.09.0
API version: 1.39
Go version: go1.10.4
Git commit: 4d60db4
Built: Wed Nov 7 00:47:51 2018
OS/Arch: windows/amd64
Experimental: false

Server: Docker Engine - Community
Engine:
Version: 18.09.0
API version: 1.39 (minimum version 1.12)
Go version: go1.10.4
Git commit: 4d60db4
Built: Wed Nov 7 00:55:00 2018
OS/Arch: linux/amd64
Experimental: false

docker run minio/minio version
Version: 2018-11-22T02:51:56Z
Release-Tag: RELEASE.2018-11-22T02-51-56Z
Commit-ID: eddf468

mc.exe version
Version: 2018-11-06T01:12:20Z
Release-tag: RELEASE.2018-11-06T01-12-20Z
Commit-id: b827fac6356f80354a1cf61f43cf5e5b60bbb511

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
Add support for hostname lookups instead of IPs in MINIO_PUBLIC_IPS

## Motivation and Context
DNS names will be resolved to their respective IPs if specified
in MINIO_PUBLIC_IPS.

Fixes #6862 

## Regression
No

## How Has This Been Tested?
Using DNS hostnames names as part of MINIO_PUBLIC_IPS list, this is simply to support the requirement presented in #6862 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

File: cmd/common-main.go [hunk 1]
```diff
@@ -245,12 +245,22 @@ func handleCommonEnvVars() {
 	minioEndpointsEnv, ok := os.LookupEnv("MINIO_PUBLIC_IPS")
 	if ok {
 		minioEndpoints := strings.Split(minioEndpointsEnv, ",")
+		var domainIPs = set.NewStringSet()
 		for i, ip := range minioEndpoints {
 			if net.ParseIP(ip) == nil {
-				logger.FatalIf(errInvalidArgument, "Unable to initialize Minio server with invalid MINIO_PUBLIC_IPS[%d]: %s", i, ip)
+				// Checking if the IP is a DNS entry.
+				addrs, err := net.LookupHost(ip)
```
Comment (vadmeste) [inline cmd/common-main.go]: Can we rename 'ip' variable to 'endpoint' ?

File: cmd/common-main.go [hunk 2]
```diff
@@ -245,12 +245,22 @@ func handleCommonEnvVars() {
 	minioEndpointsEnv, ok := os.LookupEnv("MINIO_PUBLIC_IPS")
 	if ok {
 		minioEndpoints := strings.Split(minioEndpointsEnv, ",")
+		var domainIPs = set.NewStringSet()
 		for i, ip := range minioEndpoints {
 			if net.ParseIP(ip) == nil {
-				logger.FatalIf(errInvalidArgument, "Unable to initialize Minio server with invalid MINIO_PUBLIC_IPS[%d]: %s", i, ip)
+				// Checking if the IP is a DNS entry.
```
Comment (vadmeste) [inline cmd/common-main.go:251]: Let's rename IP to endpoint

File: cmd/common-main.go [hunk 3]
```diff
@@ -245,12 +245,22 @@ func handleCommonEnvVars() {
...
+				addrs, err := net.LookupHost(ip)
+				if err != nil {
+					logger.FatalIf(err, "Unable to initialize Minio server with invalid MINIO_PUBLIC_IPS[%d]: %s", i, ip)
```
Comment (vadmeste) [inline cmd/common-main.go]: I think it is better to say 'Unable to initialize Minio server with an invalid entry (%s) found in MINIO_PUBLIC_IPS'

**Review by vadmeste — COMMENTED** [review]
Tested, LGTM, only few comments

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

**Comment by harshavardhana** [comment]
@vadmeste @nitisht addressed all the comments

**Review by nitisht — APPROVED** [review]
LGTM & Tested

**Review by vadmeste — APPROVED** [review]
LGTM & tested

**Pull request merged**